devices/YarpRobotLoggerDevice/include/BipedalLocomotion/YarpRobotLoggerDevice.h

```diff
@@ -167,10 +171,17 @@ class YarpRobotLoggerDevice : public yarp::dev::DeviceDriver,
     std::vector<std::string> m_codeStatusCmdPrefixes;
 
     std::mutex m_bufferManagerMutex;
+    std::mutex m_textLoggingPortMutex;
     robometry::BufferManager m_bufferManager;
 
     void lookForNewLogs();
     void lookForExogenousSignals();
+
+    bool initMetadata(const std::string& nameKey, const std::vector<std::string>& metadata);
+    void logData(const std::string& name,
+                                 const Eigen::VectorXd& data,
+                                 const double time);
```

Comment:
Fixed in [0888c96](https://github.com/ami-iit/bipedal-locomotion-framework/commit/0888c9631b6c934ab69fce248eae0c9587910fc7). The change of just adding Eigen::Ref<> causes issues when saving the data via buffer manager. This is because the push_back method of the buffer manager uses a template for the data and is storing the raw data of the Eigen::Ref and not the Eigen::VectorXd. The solution was discussed and implemented with the help of @S-Dafarra in which we used a lambda function to still pass the Eigen::Vector by a reference.